samples/matter/common/src/pwm_device.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "pwm_device.h"
+
+#include <lib/support/CodeUtils.h>
+
+#include <drivers/pwm.h>
+#include <logging/log.h>
+#include <zephyr.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_MATTER_LOG_LEVEL);
+
+int PWMDevice::Init(const device * aPWMDevice, uint32_t aPWMChannel, uint8_t aMinLevel, uint8_t aMaxLevel, uint8_t aDefaultLevel)
+{
+    mState      = kState_On;
```

Comment:
Formatting is not compliant with NCS. We have .clang-format file in `samples/matter` so just make sure to reformat everything. E.g. you can use my script below to fix some of compliance issues.
```
#!/bin/bash

function all_files() {
  git ls-files . | grep -v '/zap-generated/'
}

function source_files() {
	all_files | grep -P '\.(h|cpp)$'
}

# fix file names
source_files | while read f; do
  dir="$(dirname "$f")"
  old_name="$(basename "$f")"
  new_name="$(echo "$old_name" | sed -E 's|(\w)([A-Z][a-z])|\1_\2|g' | tr '[A-Z]' '[a-z]')"
  [[ "$old_name" == "$new_name" ]] && continue
  git mv "$f" "$dir/$new_name"
  sed -i "s|$old_name|$new_name|g" $(all_files)
done

# fix comments

sed -i -E 's|//(.*)$|/*\1 */|g' $(source_files) # // -> /* */
clang-format --style=file -i $(source_files)
sed -i 's|2020 Nordic Semiconductor ASA|2021 Nordic Semiconductor ASA|g' $(all_files)
```